app/src/main/java/com/jerboa/MainActivity.kt

```diff
@@ -79,11 +82,19 @@ import com.jerboa.ui.theme.JerboaTheme
 class JerboaApplication : Application() {
     private val database by lazy { AppDB.getDatabase(this) }
     val accountRepository by lazy { AccountRepository(database.accountDao()) }
-    val appSettingsRepository by lazy { AppSettingsRepository(database.appSettingsDao()) }
+    val appSettingsRepository by lazy {
+        AppSettingsRepository(database.appSettingsDao(), database.searchHistoryDao())
+    }
+    val searchHistoryRepository by lazy {
+        SearchHistoryRepository(database.searchHistoryDao(), database.appSettingsDao())
+    }
 }
 
 class MainActivity : AppCompatActivity() {
     private val siteViewModel by viewModels<SiteViewModel>()
+    private val communityListViewModel by viewModels<CommunityListViewModel>() {
+        CommunityListViewModelFactory((application as JerboaApplication).searchHistoryRepository)
```

Comment:
Why did you not make the searchhistory a viewmodel like the other tables? There's no reason to try to bundle it into another one.

Reply:
Added separate ViewModel